base/task.jl

```diff
@@ -102,12 +147,16 @@ function task_done_hook(t::Task)
                 active_repl_backend.in_eval
                 throwto(active_repl_backend.backend_task, result)
             end
-            let bt = catch_backtrace()
-                # run a new task to print the error for us
-                @schedule with_output_color(:red, STDERR) do io
-                    print(io, "ERROR (unhandled task failure): ")
-                    showerror(io, result, bt)
-                    println(io)
+            if collect_ex(t)
+                t.result=CollectedException(result, catch_backtrace())
```

Comment:
This works for `@sync` when the `@sync` block itself does not raise any exceptions - the exceptions are captured in `t.result` and finally when `wait` is called, the exceptions collected into a `CompositeException`.

I'll change the task-local setting to only decide whether to print the "unhandled task failure" or not, and to always collect the backtrace in the `Task` object.

Consider @jakebolewski 's 1000 worker scenario. If there is an exception in the `@sync` block, then only the `@sync` exception is printed, but all the 1000 async exceptions and backtraces are available as part of the `Task` objects, i.e., as long as the 1000 `@async` calls are in the context of a `@sync`, all of them will not be printed to screen.